gobblin-compaction/src/main/java/gobblin/compaction/count/HdfsRecordCountProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.count;
+
+import java.util.List;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+/**
+ * Get event count from HDFS, using given {@link FileSystem} and {@link List} of {@link Path}s.
+ */
+public interface HdfsRecordCountProvider {
```

Comment:
Also not sure why you still need this interface. Can you just use a parameter that takes the name of a class that implements `FileNameWithRecordCountFormat`, then use the `getRecordCount` method in that class?